flake.nix

```diff
@@ -111,10 +113,68 @@
         packages = {
           default = config.workspace.workspace;
           inherit (config.workspace) workspace;
+          local-telegraf = pkgs.writeShellApplication {
+            name = "local-telegraf";
+            runtimeInputs = [
+              pkgs.gnused
+              pkgs.jq
+              pkgs.yq
+              unstablePkgs.telegraf
+            ];
+            text = ''
+              # shellcheck disable=SC1091
+              source ./scripts/influx.sh
+              
+              use_influx
+              
+              WT_METRICS_DIR="$(pwd)/telegraf/metrics"
+              export WT_METRICS_DIR
+
+              # Get run id from the latest run summary or set it to ""
+              RUN_ID=$(jq --slurp --raw-output 'sort_by(.started_at|tonumber) | last | .run_id' < run_summary.jsonl)
+
+              # get tempfile for telegraf config
+              TELEGRAF_CONFIG=$(mktemp --suffix ".conf")
+              cp ./telegraf/telegraf.local.conf "$TELEGRAF_CONFIG"
+
+              sed --in-place "s/run_id = \"\"/run_id = \"$RUN_ID\"/" "$TELEGRAF_CONFIG"
+
+              echo "Running telegraf for run ID: $RUN_ID"
+              telegraf --config "$TELEGRAF_CONFIG" --once > >(tee logs/telegraf-stdout.log) 2> >(tee logs/telegraf-stderr.log >&2)
+
+              rm ./telegraf/metrics/*.influx
+            '';
+          };
           ci-telegraf = pkgs.writeShellApplication {
             name = "ci-telegraf";
-            runtimeInputs = [ pkgs.telegraf ];
-            text = "telegraf --config telegraf/runner-telegraf.conf --once > >(tee logs/telegraf-stdout.log) 2> >(tee logs/telegraf-stderr.log >&2)";
+            runtimeInputs = [
+              pkgs.gnused
+              pkgs.jq
+              pkgs.yq
+              unstablePkgs.telegraf
+            ];
+            text = ''
```

Comment:
Do Nix shell scripts exit properly on errors or should we be paranoid and add a `set` here?